src/Storages/System/StorageSystemTables.cpp

```diff
@@ -244,10 +244,32 @@ class TablesBlockSource : public SourceWithProgress
                         if (columns_mask[src_index++])
                             res_columns[res_index++]->insert(table.second->getName());
 
-                        /// Fill the rest columns with defaults
+                        auto settings = context->getSettingsRef();
                         while (src_index < columns_mask.size())
-                            if (columns_mask[src_index++])
+                        {
+                            // total_rows
+                            if (src_index == 18 && columns_mask[src_index])
+                            {
+                                auto total_rows = table.second->totalRows(settings);
+                                if (total_rows)
```

Comment:
this can be [if with initializer ](https://skebanga.github.io/if-with-initializer/)

Reply:
fixed